src/spyglass/decoding/v1/pipeline_clusterless.py

```diff
@@ -0,0 +1,283 @@
+    population for one or more decoding intervals. It also inserts the result
+    into the `DecodingOutput` merge table.
+
+    Parameters
+    ----------
+    nwb_file_name : str
+        The name of the source NWB file.
+    waveform_features_group_name : str
+        The name of the waveform features group in `UnitWaveformFeaturesGroup`.
+    position_group_name : str
+        The name of the position group in `PositionGroup`.
+    decoding_param_name : str
+        The name of the decoding parameters in `DecodingParameters`.
+    encoding_interval : str
+        The name of the interval in `IntervalList` used for encoding/training.
+    decoding_interval : Union[str, List[str]]
+        The name of the interval list (or list of names) in `IntervalList`
+        for decoding/prediction.
+    estimate_decoding_params : bool, optional
+        If True, the underlying decoder will attempt to estimate parameters.
+        Defaults to False.
+    skip_duplicates : bool, optional
+        If True, skips insertion if a matching entry already exists in
+        `ClusterlessDecodingSelection`. Defaults to True.
+    **kwargs : dict
+        Additional keyword arguments passed to the `populate` call
+        (e.g., `display_progress=True`, `reserve_jobs=True`).
+
+    Raises
+    ------
+    ValueError
+        If required upstream entries or parameters do not exist.
+
+    Examples
+    --------
+    ```python
+    # --- Example Prerequisites (Ensure these are populated) ---
+    nwb_file = 'mediumnwb20230802_.nwb'
+    wf_group = 'test_group' # From UnitWaveformFeaturesGroup tutorial
+    pos_group = 'test_group' # From PositionGroup tutorial
+    decoder_params = 'contfrag_clusterless' # From DecodingParameters tutorial
+    encode_interval = 'pos 0 valid times'
+    decode_interval = 'test decoding interval' # From Decoding tutorial
+
+    # --- Run Decoding ---
+    populate_spyglass_clusterless_decoding_v1(
+        nwb_file_name=nwb_file,
+        waveform_features_group_name=wf_group,
+        position_group_name=pos_group,
+        decoding_param_name=decoder_params,
+        encoding_interval=encode_interval,
+        decoding_interval=decode_interval,
+        estimate_decoding_params=False,
+        display_progress=True
+    )
+
+    # --- Run for multiple intervals ---
+    # decode_intervals = ['test decoding interval', 'another interval name']
+    # populate_spyglass_clusterless_decoding_v1(
+    #     nwb_file_name=nwb_file,
+    #     waveform_features_group_name=wf_group,
+    #     position_group_name=pos_group,
+    #     decoding_param_name=decoder_params,
+    #     encoding_interval=encode_interval,
+    #     decoding_interval=decode_intervals,
+    #     estimate_decoding_params=False
+    # )
+    ```
+    """
+    # --- Input Validation ---
+    if not (Nwbfile & {"nwb_file_name": nwb_file_name}):
+        raise ValueError(f"Nwbfile not found: {nwb_file_name}")
+    wf_group_key = {
```

Comment:
While relatively simple here, I do see other cases where this gets complicated. I might expect the user to pass this info as a dictionary. If that's the case, it'll make internal handling easier, and it's easier to set up the error that says, "your x key wasn't valid for table y. try running 'x & y' to check"